nucypher/policy/conditions/evm.py

```diff
@@ -154,15 +137,10 @@ def _next_endpoint(
         self, providers: Dict[int, Set[HTTPProvider]]
     ) -> Iterator[HTTPProvider]:
         """Yields the next web3 provider to try for a given chain ID"""
-        try:
-            rpc_providers = providers[self.chain]
-
-        # if there are no entries for the chain ID, there
-        # is no connection to that chain available.
-        except KeyError:
-            raise NoConnectionToChain(chain=self.chain)
+        rpc_providers = providers.get(self.chain, None)
```

Comment:
`None` is the default non-matching return value for `.get`.  However, i want to clarify how we got here - is this meant to evaluate against an empty list because all health checks failed for that chain id?

Reply:
It's possible because we no longer validate chain id against a known list of chain IDs for a domain i.e. any chain id value is valid within the definition of a condition at condition creation time. Conditions have no concept of domains, and so can't validate a chain id for `mainnet` vs `lynx`.

Remember that chains -> provider urls are obtained from the `chainlist` repos on startup and can technically contain "any" chain id mapping entry. However, if in a condition, the user provides a chain id that the domain doesn't support (eg. `chainId: 12345678987654321`) on `lynx`, then we won't have a provider for that, but we won't know until condition evaluation time when we try to get the providers for the corresponding chain ID.